### ISSUE TITLE
Expose everything

### ISSUE THREAD

**Issue by jkarni** [garnix-io/cradle#7]
I tried using `cradle` to write a helper for limiting system resources with systemd, but ran into the problem that only the `Cradle` module is exposed, and that doesn't expose the `Input` class method, or the `ProcessConfig` type. I think those should be re-exported from somewhere. But more importantly, maybe all modules should be exposed, even if with an `Internal` name? Otherwise anyone who has an unforeseen use case *has* to fork the library.

**Comment by soenkehahn** [comment]
Yeah, I think we should expose all modules. I wouldn't do that in an `Internal` module, since I think it's fine to just bump the major.minor versions often, at least in the beginning.

For writing your own `Input` or `Output` instances you do need to understand some of the internals of `cradle` so maybe we could have a warning somewhere about that.
